src/utils/gitUtils.ts

```diff
@@ -0,0 +1,143 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fse from 'fs-extra';
+import * as git from 'simple-git/promise';
+import { MessageItem, Uri } from "vscode";
+import { DialogResponses, IActionContext } from "vscode-azureextensionui";
+import { IStaticWebAppWizardContext } from "../commands/createStaticWebApp/IStaticWebAppWizardContext";
+import { ext } from "../extensionVariables";
+import { getGitApi } from "../getExtensionApi";
+import { API, CommitOptions, Ref, Repository } from "../git";
+import { hasAdminAccessToRepo, tryGetRemote, tryGetReposGetResponseData } from "./gitHubUtils";
+import { localize } from "./localize";
+
+export type GitWorkspaceState = { repo: Repository | null, dirty: boolean, remoteUrl: string | undefined; hasAdminAccess: boolean };
+export type VerifiedGitWorkspaceState = GitWorkspaceState & { repo: Repository };
+
+export async function getGitWorkspaceState(context: IActionContext & Partial<IStaticWebAppWizardContext>, uri: Uri): Promise<GitWorkspaceState> {
+    const gitWorkspaceState: GitWorkspaceState = { repo: null, dirty: false, remoteUrl: undefined, hasAdminAccess: false };
+    const gitApi: API = await getGitApi();
+    const repo: Repository | null = await gitApi.openRepository(uri);
+
+    if (repo) {
+        const originUrl: string | undefined = await tryGetRemote(uri.fsPath);
+        gitWorkspaceState.repo = repo;
+        gitWorkspaceState.dirty = !!(repo.state.workingTreeChanges.length || repo.state.indexChanges.length);
+        gitWorkspaceState.remoteUrl = originUrl;
+        gitWorkspaceState.hasAdminAccess = originUrl ? hasAdminAccessToRepo(await (tryGetReposGetResponseData(context, originUrl))) : false;
+    }
+
+    return gitWorkspaceState;
+}
+
+/* Function used to enforce all our opinionated requirements for creating a Static Web App */
+export async function verifyGitWorkspaceForCreation(context: IActionContext, gitWorkspaceState: GitWorkspaceState, uri: Uri): Promise<VerifiedGitWorkspaceState> {
+    const gitFolderName: string = '.git';
+    if ((await fse.readdir(uri.fsPath)).filter(file => file !== gitFolderName).length === 0) {
+        // if the git repo is empty, git push will fail and create an empty GitHub repo so throw an error here
+        throw new Error(localize('emptyWorkspace', 'Cannot create a Static Web App with an empty workspace.'));
+    }
+
+    if (!gitWorkspaceState.repo) {
+        const gitRequired: string = localize('gitRequired', 'A GitHub repository is required to create a Static Web App. Would you like to initialize your project as a git repository and create a GitHub remote?');
```

Comment:
I liked Matt's suggestion to switch up the wording here so it isn't in the form of a question since people don't have a choice